modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -1014,27 +1050,75 @@ class NDB_Form_Statistics extends NDB_Form
             $projects[$key] = $value;
         }
 
-        if (isset($_REQUEST['ReliabilityProject'])
-            && $_REQUEST['ReliabilityProject'] != ''
-        ) {
-            $currentProject = $_REQUEST['ReliabilityProject'];
+        if ($_REQUEST['ReliabilityProject'] ?? '') {
+            $currentProject = htmlspecialchars($_REQUEST['ReliabilityProject']);
             $this->tpl_data['CurrentProject']
                 = array(
                    'ID'   => $currentProject,
                    'Name' => $projects[$currentProject],
                   );
             $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] =$_REQUEST['ReliabilityProject'];
+            $this->params['pid'] = htmlspecialchars($_REQUEST['ReliabilityProject']);
         } else {
             $Param_Project ='';
         }
 
-        //SITES
-        if (isset($_REQUEST['ReliabilitySite'])
-            && $_REQUEST['ReliabilitySite'] != ''
+        //SUBPROJECTS
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
+
+        foreach ($subprojects as $key=>$val) {
+            $subprojList .= $key.",";
+        }
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubproject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if ($_REQUEST['ReliabilitySubproject'] ?? ''
+            && in_array(
+                $_REQUEST['ReliabilitySubproject'],
+                array_keys($subprojects),
+                true
+            )
         ) {
+            $RS = 'ReliabilitySubproject';
+            $currentSubproject = htmlspecialchars($_REQUEST[$RS]);
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubproject,
+                   'Name' => $subprojects[$currentSubproject],
+                  );
+            $subproject_query     = "AND (s.SubprojectID IS NULL ";
+            $subproject_query    .= "OR s.SubprojectID=:spid)";
+            $this->params['spid'] =$currentSubproject;
+        } else {
+            $subproject_query ="";
+        }
+        if ($_REQUEST['ReliabilitySubproject'] ?? ''
+            && !in_array(
+                $_REQUEST['ReliabilitySubproject'],
+                array_keys(
+                    $subprojects,
+                    true
+                )
```

Comment:
same question about why this isn't in the existing if/else blocks, instead of adding a new (but opposite) if/else block.